internal/cluster/cluster.go

```diff
@@ -628,3 +643,72 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 	}
 	return nil
 }
+
+func (m *Manager) DeleteClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	log := logutil.FromContext(ctx, m.log)
+	files, err := objectHandler.ListObjectsByPrefix(ctx, fmt.Sprintf("%s/logs/", c.ID))
+	if err != nil {
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	var failedToDelete []string
+	for _, file := range files {
+		log.Debugf("Deleting cluster %s S3 log file: %s", c.ID.String(), file)
+		if err := objectHandler.DeleteObject(ctx, file); err != nil {
+			m.log.WithError(err).Errorf("failed deleting s3 log %s", file)
+			failedToDelete = append(failedToDelete, file)
+		}
+	}
+
+	if len(failedToDelete) > 0 {
+		return common.NewApiError(
+			http.StatusInternalServerError,
+			errors.Errorf("failed to delete s3 logs: %q", failedToDelete))
+	}
+	return nil
+}
+
+func (m *Manager) DeleteClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	var failedToDelete []string
+	for _, name := range S3FileNames {
+		fileName := fmt.Sprintf("%s/%s", c.ID, name)
+		if err := objectHandler.DeleteObject(ctx, fileName); err != nil {
+			m.log.WithError(err).Errorf("failed deleting s3 file %s", fileName)
+			failedToDelete = append(failedToDelete, fileName)
+		}
+	}
+
+	if len(failedToDelete) > 0 {
+		return common.NewApiError(
+			http.StatusInternalServerError,
+			errors.Errorf("failed to delete s3 files: %q", failedToDelete))
+	}
+	return nil
+}
+
+func (m Manager) PermanentClustersDeletion(ctx context.Context, olderThen strfmt.DateTime, objectHandler s3wrapper.API) error {
```

Comment:
General comment that i missed from the review - when working with monitors if you fail an operation on one cluster you probably need to continue and not return an error - it's relevant for all the functions that you called.